ocw/lib/provider.py

```diff
@@ -62,5 +64,11 @@ def log_dbg(self,  message: str, *args: object):
             message = message.format(*args)
         self.logger.debug("[%s] %s", self._namespace, message)
 
-    def cmd_exec(self, cmd):
-        return subprocess.call(cmd.split())
+    def cmd_exec(self, cmd, aditional_env=None, expected_returncode=0):
```

Comment:
1. I didn't get how `check_returncode()` will "know" about which code to check 
2. I assume that `check_returncode()` will throw exception when code does not much . It is not always suitable sometimes we want to not raise exception but just log warn message or have several behaviors depending what was returned

Reply:
done